optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/bi/DroolsBiCondition.java

```diff
@@ -53,6 +57,14 @@ public DroolsBiCondition(DroolsMetadata<Object, A> aMetadata, DroolsMetadata<Obj
         this.bMetadata = bMetadata;
     }
 
+    public DroolsBiCondition(Declaration<DroolsLogicalTuple> aVariableDeclaration,
+            Function<Declaration<DroolsLogicalTuple>, PatternDSL.PatternDef<DroolsLogicalTuple>> patternProvider) {
+        // We share both the declaration and the pattern, as the data is all coming from the same DroolsLogicalTuple.
```

Comment:
style nitpick: use active voice (instead of "we") to be more compact:
=> // Share both the declaration and the pattern, ...